Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/detail/ConnectedComponents.cuh

```diff
@@ -97,6 +119,42 @@ __global__ void labelConnectedComponents(std::size_t numEdges,
   } while (__syncthreads_or(changed));
 }
 
+/// Hooking-kernel for implementing the FastSV loop on the host
+template <typename TEdge, typename TLabel>
+__global__ void hookEdges(std::size_t numEdges, const TEdge *sourceEdges,
+                          const TEdge *targetEdges, const TLabel *labels,
+                          TLabel *labelsNext, int *globalChanged) {
+  int i = threadIdx.x + blockIdx.x * blockDim.x;
+  if (i >= numEdges) {
+    return;
+  }
+
+  bool changed = false;
+  hookEdgesImpl(i, sourceEdges, targetEdges, labels, labelsNext, changed);
+
+  if (__syncthreads_or(changed) && threadIdx.x == 0) {
+    atomicOr(globalChanged, true);
```

Comment:
Although I agree that this is the cleaner way to do this, you can also just set `*globalChanged = true` non-atomically; this will eventually become consistent.

Reply:
Alternatively you can use warp voting:

```suggestion
bool any_in_warp_changed = __any_sync(__activemask(), changed);
if ((threadIdx.x & 0x1f) == 0 && any_in_warp_changed) {
    *globalChanged = true;
}
```

Reply:
Hmm I think I find it somewhat easier to not do the early return, but I removed the atomic access.

Reply:
The small advantage with the warp voting is you remove the block-wide synchronization point, but I would suspect that we are talking about very small performance effects here.